File: frontend/src/types/history.ts
export interface FinancialSnapshotResponse {
  timestamp: string
  total_income: number
  total_expenses: number
  net_cash_flow: number
  total_account_balance: number
  total_goal_progress: number
  total_debt: number
  net_worth: number
  health_score: number
  health_status: string
}

export type TrendDirection = 'Improving' | 'Declining' | 'Stable' | 'Insufficient Data'
export type OverallMomentum = 'Positive' | 'Negative' | 'Stable' | 'Insufficient Data'

export interface MetricTrend {
  direction: TrendDirection
  change: number
}

export interface TrendSummary {
  net_worth: MetricTrend
  cash_flow: MetricTrend
  income: MetricTrend
  expenses: MetricTrend
  health_score: MetricTrend
  overall_momentum: OverallMomentum
}

File: src/api/schemas/history.py
"""API schemas for financial history endpoints."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class FinancialSnapshotResponse(BaseModel):
    """Serialized representation of a historical financial snapshot."""

    model_config = ConfigDict(from_attributes=True)

    timestamp: datetime
    total_income: float
    total_expenses: float
    net_cash_flow: float
    total_account_balance: float
    total_goal_progress: float
    total_debt: float
    net_worth: float
    health_score: int
    health_status: str
